Add sorting parameters to get_videos_for_ids

gwprice opened the pull request:
Please review @nasthagiri @BenjiLee

BenjiLee commented on edxval/api.py:
```diff
@@ -234,17 +250,30 @@ def get_videos_for_course(course_id):
     Args:
         edx_video_ids (list)
+        sort_field (VideoSortField)
+        sort_direction (SortDirection)
```
sort_direction --> sort_dir

gwprice replied:
Will do

BenjiLee commented on edxval/api.py:
```diff
@@ -234,17 +250,30 @@ def get_videos_for_course(course_id):
     videos = Video.objects.filter(edx_video_id__in=edx_video_ids)
+    if sort_field:
+        # Refining by edx_video_id ensures a total order
+        videos = videos.order_by(sort_field.value, "edx_video_id")
```
Is the default sort ascending?

gwprice replied:
Yes, order_by sorts ascending unless the field is prefixed with `-`

BenjiLee commented on edxval/tests/test_api.py:
```diff
@@ -334,6 +334,45 @@ def test_get_videos_for_ids_duplicates(self):
+            [star_id, fish_id]
+        )
+        check_sort(
+            VideoSortField.duration,
```
I believe fish and star have equal durations.

gwprice replied:
Yes, and the refinement by edx_video_id guarantees this ordering.

nasthagiri commented on edxval/tests/test_api.py:
```diff
@@ -334,6 +334,45 @@ def test_get_videos_for_ids_duplicates(self):
+            VideoSortField.duration,
+            SortDirection.desc,
+            [fish_id, star_id]
+        )
```
Can you add a test for sorting by `edx_video_id`?  Just to validate that adding `edx_video_id` to the sorting by `edx_video_id` works.

nasthagiri replied:
Also, do any of these tests verify the total order refinement that you've implemented?  That is, a test that needs `edx_video_id` to break a tie.

gwprice replied:
Will do. As noted above, the videos have the same duration, so that case is covered.

nasthagiri commented:
:+1: pending above comments.  looks good overall.

gwprice commented:
Updated with a commit to address feedback. I refactored the tests pretty substantially and added another video to make the tests a little bit more rigorous.

nasthagiri commented on edxval/tests/test_api.py:
```diff
@@ -334,6 +334,43 @@ def test_get_videos_for_ids_duplicates(self):
+                """Assert that the given videos match the expected ids"""
+                actual_videos = api.get_videos_for_ids(
+                    # Make sure it's not just returning the order given
+                    sorted(expected_ids),
```
why sorted and not reversed? with sorted, it may still remain as the same order.

gwprice replied:
Will change

nasthagiri commented on edxval/tests/test_api.py:
```diff
@@ -334,6 +334,43 @@ def test_get_videos_for_ids_duplicates(self):
+        star_id = constants.VIDEO_DICT_STAR["edx_video_id"]
+        other_id = "other-video"
+        Video.objects.create(**constants.VIDEO_DICT_STAR)
+        Video.objects.create(
```
To be consistent with the other tests in this file, consider using another already existing video in `constants` or moving the declaration of this video to the `constants` file.

gwprice replied:
The constants file is already very crowded, and this video is specifically crafted to sort interestingly compared with the other two. I think that's clearer if it's defined here; I can certainly add a comment to that effect.

BenjiLee commented on edxval/tests/test_api.py:
```diff
@@ -334,6 +334,43 @@ def test_get_videos_for_ids_duplicates(self):
+        )
+
+        def check_sort(sort_field, asc_expected_ids):
+            """
```
What does asc_expected_ids mean? Does it mean we are expecting an ascending list of ids?

gwprice replied:
It's the list of expected ids for the ascending case

gwprice replied:
I'm changing it to `expected_ids_for_asc`

gwprice commented:
Updated with two more commits to address feedback

BenjiLee commented:
:+1:

nasthagiri commented:
:+1:
